### ISSUE TITLE
Race condition?

### ISSUE THREAD

**Issue by drbyte** [jeromejaglale/laravel-github-webhook#1]
Hi @jeromejaglale ,
Thanks for your article discussing code deployment, and for this repo of associated code snippets.

Question:
Have you run into a situation where the deploy fails because the deploy script files themselves are still "open" because the script is executing?

Are you still using this approach to deploy your own apps? Anything you'd change now that a couple years have gone by?

Thanks for any insights!

**Comment by jeromejaglale** [comment]
Hi @drbyte ,

> Have you run into a situation where the deploy fails because the deploy script files themselves are still "open" because the script is executing?

You mean when a deployment is already happening? Yes. I usually do an extra small push a bit later.

> Are you still using this approach to deploy your own apps? Anything you'd change now that a couple years have gone by?

Yes. No, it works well enough for my purposes.

**Comment by drbyte** [comment]
Ya, I worry that composer won’t update the symfony/process and other vendor files if they’re actually running the deploy.

**Comment by jeromejaglale** [comment]
I don't think this will be a problem. I'm planning to do just that next week, I'll let you know :)
